concrete-boolean/src/engine/bootstrapping/cpu.rs

```diff
@@ -0,0 +1,301 @@
+        let mut fftw_engine = FftwEngine::new(()).unwrap();
+
+        let bootstrap_key: LweBootstrapKey32 = engine.create_lwe_bootstrap_key(
+            &cks.lwe_secret_key,
+            &cks.glwe_secret_key,
+            cks.parameters.pbs_base_log,
+            cks.parameters.pbs_level,
+            var_rlwe,
+        )?;
+
+        let fourier_bsk = fftw_engine.convert_lwe_bootstrap_key(&bootstrap_key)?;
+
+        // Convert the GLWE secret key into an LWE secret key:
+        let big_lwe_secret_key =
+            engine.transform_glwe_secret_key_to_lwe_secret_key(cks.glwe_secret_key.clone())?;
+
+        // convert into a variance for lwe context
+        let var_lwe = Variance(cks.parameters.lwe_modular_std_dev.get_variance());
+        // creation of the key switching key
+        let ksk = engine.create_lwe_keyswitch_key(
+            &big_lwe_secret_key,
+            &cks.lwe_secret_key,
+            cks.parameters.ks_level,
+            cks.parameters.ks_base_log,
+            var_lwe,
+        )?;
+
+        Ok(Self {
+            bootstrapping_key: fourier_bsk,
+            key_switching_key: ksk,
+        })
+    }
+}
+
+impl BooleanServerKey for CpuServerKey {
+    fn lwe_size(&self) -> LweSize {
+        self.bootstrapping_key.input_lwe_dimension().to_lwe_size()
+    }
+}
+
+/// Memory used as buffer for the bootstrap
+///
+/// It contains contiguous chunk which is then sliced and converted
+/// into core's View types.
+#[derive(Default)]
+struct Memory {
+    elements: Vec<u32>,
+}
+
+impl Memory {
+    /// Returns a tuple with buffers that matches the server key.
+    ///
+    /// - The first element is the accumulator for bootstrap step.
+    /// - The second element is a lwe buffer where the result of the of the bootstrap should be
+    ///   written
+    fn as_buffers(
+        &mut self,
+        engine: &mut DefaultEngine,
+        server_key: &CpuServerKey,
+    ) -> (GlweCiphertextView32, LweCiphertextMutView32) {
+        let num_elem_in_accumulator = server_key
+            .bootstrapping_key
+            .glwe_dimension()
+            .to_glwe_size()
+            .0
+            * server_key.bootstrapping_key.polynomial_size().0;
+        let num_elem_in_lwe = server_key
+            .bootstrapping_key
+            .output_lwe_dimension()
+            .to_lwe_size()
+            .0;
+        let total_elem_needed = num_elem_in_accumulator + num_elem_in_lwe;
+
+        let all_elements = if self.elements.len() < total_elem_needed {
+            self.elements.resize(total_elem_needed, 0u32);
+            self.elements.as_mut_slice()
+        } else {
+            &mut self.elements[..total_elem_needed]
+        };
+
+        let (accumulator_elements, lwe_elements) =
+            all_elements.split_at_mut(num_elem_in_accumulator);
+        accumulator_elements
+            [num_elem_in_accumulator - server_key.bootstrapping_key.polynomial_size().0..]
+            .fill(PLAINTEXT_TRUE);
```

Comment:
So the accumulator is a GLWE ciphertext with mask and body being polynomials. (A_1, ...A_n, B) where n = GlweSize and Each A_x and B are polynomials of size polynomial size.

In concrete a GLWE ciphertext is stored as a flat Vec where first, we have the elements of the mask, then the body.

And so what we do here is a trivial encryption of a GLWE where all elements in the masks are zeros, and the body
directly contains the plaintext (with no noise as its a trivial encryption). That's why we only init a part of the buffer,
whe only initialize the body part.

`num_elem_in_accumulator` is defined as 
```rust
  let num_elem_in_accumulator = server_key
            .bootstrapping_key
            .glwe_dimension()
            .to_glwe_size()
            .0
            * server_key.bootstrapping_key.polynomial_size().0;
```
I don't think the LHS part can ever be less than the RHS, the Lhs since
`bootstrapping_key .glwe_dimension() .to_glwe_size()` will at least be one since `glwe_size = glwe_dimension + 1`. 
so if concrete allows `glwe_dimension = 0` then we would have our Lhs = Rhs